src/components/sync-configurator/BaseSyncConfigurator.jsx

```diff
@@ -173,4 +173,4 @@ class BaseSyncConfigurator extends React.Component {
     }
 }
 
-export default withRouter(withSnackbar(withStyles(styles)(BaseSyncConfigurator)));
+export default withSnackbar(withRouter(withStyles(styles)(BaseSyncConfigurator)));
```

Comment:
I guess that solves the "Warning contextType". Did you understand the details?

Reply:
More or less I understand the cause but not the details. It's something about the way withRouter builds the HOC, it doesn't like having contextType set inside it.